### PR TITLE
Feature/entity api

### PR DESCRIPTION
**Pull request by edsardachuk**

### CODE REVIEW

File: docs/backend-specialist/2.3-entity-api.md [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+The same applies to config entity types. Sometimes, simple configuration forms that save values to a single config file are not enough, especially when you need to create different configurations of the same structure/type.
+
+For example, if you need a system like **CompanyType**, it's not ideal to store it alongside content like **Countries** or **Tags**. System configurations should be static, and developers should refer to entities like **CompanyType** by machine name (available in custom entity types), not by term ID.
+
+## Creating a Custom Entity Type
+
+To create a custom entity type, you need a custom module and a set of interconnected files with consistent syntax and machine names. To simplify the process, tools for automatic file generation are available, such as:
+
+- **Drupal Console (deprecated)**
+- **Drush 12 (generate command)**
```
Comment (samuelmolinski) [inline docs/backend-specialist/2.3-entity-api.md]: Remove the version reference, or specify that Drush v11-13 supports Drupal 10. https://www.drush.org/13.2.0/install/#drupal-compatibility

Comment (samuelmolinski) [reply docs/backend-specialist/2.3-entity-api.md]: I would keep in mind that we are focused on Drupal 10 Certification.

File: docs/backend-specialist/2.3-entity-api.md [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+
+
+---
+
+## Questions:
+
+
+::: code-group
+``` [Question 1]
+You have an existing custom Content Entity Type on the Live site. You need to add new base field to this entity type. How will you do it?
```
Comment (samuelmolinski) [inline docs/backend-specialist/2.3-entity-api.md]: Great questions. We need to limit the question a bit more. "How do we programmatically add a new field in an update hook?" My reflex is to make the changes in the UI and push configuration changes.

File: docs/backend-specialist/2.3-entity-api.md [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+You have custom Content Entity Type. You have added new operation into it's declaration. How to apply your changes?
+```
+
+``` [Answer]
+// We should use hook_update as well.
+
+/**
+ * Add new operation to entity.
+ */
+function custom_module_update_1002() {
```
Comment (samuelmolinski) [inline docs/backend-specialist/2.3-entity-api.md:458]: This is my first time looking at "adding an operation to an entity type," so I might be the perfect audience. Could you give me more context about what is happening in the code sample below? Is the"operation" previously defined? What is happening on each line? I understand the first two lines, but it seems abit circular. Lets talk about this one later.

File: docs/backend-specialist/2.3-entity-api.md [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+    return $fields;
+  }
+```
+
+More info on possible configurations in annotation: https://www.drupal.org/docs/drupal-apis/entity-api/introduction-to-entity-api-in-drupal-8
+
+## Bundles
+
+Sometimes, a content entity type needs bundles. For example, a **Node** is a content entity, while its bundles are **NodeType** config entities.
+
```
Comment (selwynpolit) [inline docs/backend-specialist/2.3-entity-api.md:291]: Want to add something like this?
A content entity type can have multiple bundles, which are essentially subtypes of the entity. For example, the Node entity type has bundles like Article and Page.  
So:
* Content Entity Type: A general category of content, such as Node, User, or Taxonomy Term.
* Bundle: A specific subtype of a content entity type, such as Article or Page for the Node entity type.

File: docs/backend-specialist/2.3-entity-api.md [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+## Why Create Custom Entity Types Instead of Using Nodes for Everything?
+
+For simple websites, Drupal's default entities are usually sufficient. However, for complex systems, a well-architected solution may involve developing custom configuration and content entity types.
+
+The **Node** entity comes with many content-related fields and a lot of built-in functionality. For a clean architecture, it’s better to create a new content entity type with only the required fields. This allows developers to easily customize the entity, add Views handlers, and tweak the list builder and `save()` method without relying on hooks.
+
+The same applies to config entity types. Sometimes, simple configuration forms that save values to a single config file are not enough, especially when you need to create different configurations of the same structure/type.
+
+For example, if you need a system like **CompanyType**, it's not ideal to store it alongside content like **Countries** or **Tags**. System configurations should be static, and developers should refer to entities like **CompanyType** by machine name (available in custom entity types), not by term ID.
+
```
Comment (selwynpolit) [inline docs/backend-specialist/2.3-entity-api.md:46]: More explanation?
In this example, `CompanyType` is a configuration entity type, which is static and should be referred to by its machine name. This is in contrast to content entities like `Countries` or `Tags`, which are more dynamic and typically referred to by term ID.

File: docs/backend-specialist/2.3-entity-api.md (hunk 1 above)
Comment (edsardachuk) [reply docs/backend-specialist/2.3-entity-api.md]: changed

File: docs/backend-specialist/2.3-entity-api.md (hunk 2 above)
Comment (edsardachuk) [reply docs/backend-specialist/2.3-entity-api.md]: The question says "add a new base field". Base field definitions cannot be added via UI, only programmatically. But I improved the question. Thanks.

File: docs/backend-specialist/2.3-entity-api.md (hunk 3 above)
Comment (edsardachuk) [reply docs/backend-specialist/2.3-entity-api.md:458]: I added more comments

File: docs/backend-specialist/2.3-entity-api.md (hunk 4 above)
Comment (edsardachuk) [reply docs/backend-specialist/2.3-entity-api.md:291]: Added

File: docs/backend-specialist/2.3-entity-api.md (hunk 5 above)
Comment (edsardachuk) [reply docs/backend-specialist/2.3-entity-api.md:46]: thanks, improved

**Pull request merged**